Newsletters: add 210 (2022-07-27) (#816)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2022-07-27-newsletter.md

  other wallets have long included support for signing and verifying
  arbitrary messages when the key used to sign them corresponds to a
  P2PKH address.  Bitcoin Core doesn't support signing or verifying
  arbitrary messages for any other address type, including addresses
  covering single-sig P2SH-P2WPKH, native P2WPKH, and P2TR outputs.  A
  previous proposal, [BIP322][], to provide [fully generic message
  signing][topic generic signmessage] that could work with any script
  has not yet [been merged][bitcoin core #24058] into Bitcoin Core or
  added to any other popular wallet of which we're aware.

    This week, Ali Sherief [proposed][sherief gsm] that the same message
    signing algorithm used for P2WPKH also be used for other output
    types.  For verification, programs should infer how to derive the
    key (if necessary) and verify the signature using the address type.
    E.g., when provided a [bech32][topic bech32] address with a 20 byte
    data element, assume it's for a P2WPKH output.

    Developer Peter Gray of Coinkite [noted][gray cc] that ColdCard
    wallets already create signatures in this way and developer Craig

>     Raw of Sparrow Wallet [said][raw sparrow] that wallet is able to

Reviewer: ```suggestion
    Raw [said][raw sparrow] that Sparrow Wallet wallet is able to
```

Reviewer: Indeed

    validate them in addition to also following the [BIP137][]
    validation rules and a slightly different set of rules implemented
    in Electrum.

    Sherief is planning to write a BIP specifying the behavior.

> - **Proof of micro-burn:** several developers [discussed][pomb] how to
>   allow a single onchain transaction that destroys bitcoins ("burns"
>   bitcoins) be used in several different places for proving a subset of
>   those bitcoins had been burned.  To extend an example of Ruben

Reviewer 2: This seems a bit circular to me, like "burn bitcoins to prove bitcoins have been burned"
Perhaps mention that both use cases and methods are discussed?
```suggestion
- **Proof of micro-burn:** several developers [discussed][pomb]
  use cases and designs of onchain transactions that destroy bitcoins
  ("burn" bitcoins) as a proof of resource consumption. To extend an example use case of Ruben
```

  covering single-sig P2SH-P2WPKH, native P2WPKH, and P2TR outputs.  A
  previous proposal, [BIP322][], to provide [fully generic message
  signing][topic generic signmessage] that could work with any script
  has not yet [been merged][bitcoin core #24058] into Bitcoin Core or
  added to any other popular wallet of which we're aware.

    This week, Ali Sherief [proposed][sherief gsm] that the same message
    signing algorithm used for P2WPKH also be used for other output
    types.  For verification, programs should infer how to derive the
    key (if necessary) and verify the signature using the address type.
    E.g., when provided a [bech32][topic bech32] address with a 20 byte
    data element, assume it's for a P2WPKH output.

    Developer Peter Gray of Coinkite [noted][gray cc] that ColdCard
    wallets already create signatures in this way and developer Craig
    Raw of Sparrow Wallet [said][raw sparrow] that wallet is able to
    validate them in addition to also following the [BIP137][]
    validation rules and a slightly different set of rules implemented
    in Electrum.

    Sherief is planning to write a BIP specifying the behavior.

> - **Proof of micro-burn:** several developers [discussed][pomb] how to

Reviewer 3: "how to allow a single onchain transaction... be used" sounds awkward to me

  [`nMinimumChainWork`][Bitcoin Core #9053] help filter out low-difficulty chain attacks.

## Releases and release candidates

*New releases and release candidates for popular Bitcoin infrastructure
projects.  Please consider upgrading to new releases or helping to test
release candidates.*

FIXME:harding

- BTCPay Server 1.6.2

## Notable code and documentation changes

*Notable changes this week in [Bitcoin Core][bitcoin core repo], [Core
Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*

- [Bitcoin Core #25351][] rpc, wallet: Scan mempool after import - Second attempt FIXME:Xekyo

- [Core Lightning #5370][] reimplements the `commando` plugin and makes
  it a built-in part of CLN.  Commando allows a node to receive commands
  from authorized peers using LN messages.  Peers are authorized using
  *runes*, which is a custom CLN protocol based on a simplified version

>   of [macaroons][].  Although Commando is now built into CLN, it's only

Reviewer 3: ```suggestion
  of [macaroons][].  Although commando is now built into CLN, it's only
```

  operable if a user creates rune authentication tokens.  For additional
  information, see CLN's manual pages for [commando][] and [commando-rune][].

- [BOLTs #1001][] recommends that nodes who advertise a change to their
  payment forwarding policies continue accepting payments received
  using the old policies for about 10 minutes.  This prevents payments
  from failing just because the sender hasn't heard about a recent
  policy update.  See [Newsletter #169][news169 cln4806] for the example
  of an implementation adopting a rule like this.

{% include references.md %}
{% include linkers/issues.md v=2 issues="25351,5370,1001,24058,9053" %}

> [commando]: https://github.com/rustyrussell/lightning/blob/2e13b72f55080be07ea68de77976eb990a043f5d/doc/lightning-commando.7.md

Reviewer 3: ```suggestion
[commando]: https://github.com/ElementsProject/lightning/blob/master/doc/lightning-commando.7.md
```

Author: I prefer to link to a specific commit so that the link doesn't break if the file is later moved or deleted.

> [commando-rune]: https://github.com/rustyrussell/lightning/blob/2e13b72f55080be07ea68de77976eb990a043f5d/doc/lightning-commando-rune.7.md

Reviewer 3: ```suggestion
[commando-rune]: https://github.com/ElementsProject/lightning/blob/master/doc/lightning-commando-rune.7.md
```

  has not yet [been merged][bitcoin core #24058] into Bitcoin Core or
  added to any other popular wallet of which we're aware.

    This week, Ali Sherief [proposed][sherief gsm] that the same message
    signing algorithm used for P2WPKH also be used for other output
    types.  For verification, programs should infer how to derive the
    key (if necessary) and verify the signature using the address type.
    E.g., when provided a [bech32][topic bech32] address with a 20 byte
    data element, assume it's for a P2WPKH output.

    Developer Peter Gray of Coinkite [noted][gray cc] that ColdCard
    wallets already create signatures in this way and developer Craig
    Raw of Sparrow Wallet [said][raw sparrow] that wallet is able to
    validate them in addition to also following the [BIP137][]
    validation rules and a slightly different set of rules implemented
    in Electrum.

    Sherief is planning to write a BIP specifying the behavior.

- **Proof of micro-burn:** several developers [discussed][pomb] how to
  allow a single onchain transaction that destroys bitcoins ("burns"
  bitcoins) be used in several different places for proving a subset of

>   those bitcoins had been burned.  To extend an example of Ruben
>   Somsen's [from the thread][somsen hashcash], the idea would be to

Reviewer 4: I think that might be a bit too possessive:

```suggestion
  those bitcoins had been burned.  To extend an example by Ruben
  Somsen [from the thread][somsen hashcash], the idea would be to
```

  bitcoins had been burned, providing the type of anti-spam protection
  originally envisioned as a benefit of [hashcash][].

    Several solutions were discussed using merkle trees, although one
    respondent suggested that the small amounts involved suggest that
    having participants trust (or partially trust) a centralized third
    party may be a reasonable way to avoid unnecessary complexity.

## Selected Q&A from Bitcoin Stack Exchange

*[Bitcoin Stack Exchange][bitcoin.se] is one of the first places Optech
contributors look for answers to their questions---or when we have a
few spare moments to help curious or confused users.  In
this monthly feature, we highlight some of the top-voted questions and
answers posted since our last update.*

{% comment %}<!-- https://bitcoin.stackexchange.com/search?tab=votes&q=created%3a1m..%20is%3aanswer -->{% endcomment %}
{% assign bse = "https://bitcoin.stackexchange.com/a/" %}

- [Why do invalid signatures in `OP_CHECKSIGADD` not push to the stack?]({{bse}}114446)
  Chris Stewart asks why "if an invalid signature is found, the interpreter fails
  execution rather than continuing". Pieter Wuille explains that this behavior,

>   defined in BIP340-342, is designed so that support for batch validation of
>   [schnorr signatures][topic schnorr signatures] is possible in the future.

Reviewer 4: ```suggestion
  defined in BIP340-342, is designed to support batch validation of
  [schnorr signatures][topic schnorr signatures] in the future.
```

>   Andrew Chow give an additional reason for the behavior, noting that
>   certain malleability concerns are also mitigated with this approach.

Reviewer 4: ```suggestion
  Andrew Chow gives an additional reason for the behavior, noting that
  certain malleability concerns are also mitigated by this approach.
```
